pydantic/_internal/_known_annotated_metadata.py

```diff
@@ -168,6 +157,25 @@ def expand_grouped_metadata(annotations: Iterable[Any]) -> Iterable[Any]:
             yield annotation
 
 
+@lru_cache
+def _get_at_to_constraint_map() -> dict[type, str]:
+    """Return a mapping of annotated types to constraints.
+
+    Stored in this function because we don't allow global imports of annotated types, but we cache for performance.
```

Comment:
May also want to mention that the main reason for doing this in a cached function is not so much for performance, but so that we can define it in one place rather than copy/pasting the definition. We need it to be in a function in order to defer the import, and we cache the result of that function call because .. why not